build_tools/cmake/iree_copts.cmake

```diff
@@ -138,47 +138,19 @@ set(IREE_TEST_COPTS "${ABSL_TEST_COPTS}")
 # Sanitizer configurations
 #-------------------------------------------------------------------------------
 
-include(CheckCXXCompilerFlag)
 if(${IREE_ENABLE_ASAN})
-  set(CMAKE_REQUIRED_FLAGS "-fsanitize=address")
-  check_cxx_compiler_flag(-fsanitize=address COMPILER_SUPPORTS_ASAN)
-  unset(CMAKE_REQUIRED_FLAGS)
-  if(${COMPILER_SUPPORTS_ASAN})
-    list(APPEND IREE_DEFAULT_COPTS "-fsanitize=address")
-    if(ANDROID)
-      list(APPEND IREE_DEFAULT_LINKOPTS "-fuse-ld=gold")
-    endif()
-  else()
-    message(FATAL_ERROR "The compiler does not support address sanitizer "
-                        "or is missing configuration for address sanitizer")
```

Comment:
> Don't check for compiler support for -fsanitize= flags.
If the user explicitly enables a sanitizer, just use it.
This only makes a difference to the error message, should
not make any change to what succeeds/fails.

Part of this error check was meant to give a more useful message on MSVC where different flags might be used: https://github.com/google/iree/pull/2832#discussion_r469345847. However, from another look at https://devblogs.microsoft.com/cppblog/addresssanitizer-asan-for-windows-with-msvc/ and https://devblogs.microsoft.com/cppblog/asan-for-windows-x64-and-debug-build-support/, it looks like the same flag is used.